### PR TITLE
MM-47066 : Migrate "components/suggestion/suggestion_date.jsx" to Typescript

### PR DESCRIPTION
**Pull request by BediNimret**
#### Summary
MM-47066:Migrate "components/suggestion/suggestion_date.jsx" to Typescript

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/21097
Fixes https://mattermost.atlassian.net/browse/MM-47066

#### Related Pull Requests
None 

#### Screenshots
None

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/suggestion/suggestion_list.d.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import React from 'react';
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 interface SuggestionItem {}
 
-interface Props {
```
Comment (M-ZubairAhmed) [inline components/suggestion/suggestion_list.d.ts:12]: we will get rid of this file when we migrate `suggestion_list` so i would recommend defining the props in the date component

File: utils/constants.tsx [hunk 2]
```diff
@@ -5,10 +5,10 @@
 
 import keyMirror from 'key-mirror';
 
-import Permissions from 'mattermost-redux/constants/permissions';
-
```
Comment (M-ZubairAhmed) [inline utils/constants.tsx:9]: can you revert this change

File: components/suggestion/suggestion_date.tsx [hunk 3]
```diff
@@ -1,20 +1,13 @@
...
 
 import Popover from 'components/widgets/popover';
 
-export default class SuggestionDate extends React.PureComponent {
-    static propTypes = {
-        onCompleteWord: PropTypes.func.isRequired,
-        items: PropTypes.array.isRequired,
-        terms: PropTypes.array.isRequired,
-        components: PropTypes.array.isRequired,
-        matchedPretext: PropTypes.array.isRequired,
```
Comment (M-ZubairAhmed) [inline components/suggestion/suggestion_date.tsx:15]: lets define what all props are actually used in this component and declare the props interface with them

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by m1lt0n — APPROVED** [review]

File: components/suggestion/suggestion_list.d.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ interface Props {
     items: SuggestionItem[];
     terms: string[];
     selection: string;
-    components: Array<React.Component<{item: SuggestionItem}>>;
+    components: ReactComponent<Array<React.Component<{item: SuggestionItem}>>>;
```
Comment (M-ZubairAhmed) [inline components/suggestion/suggestion_list.d.ts:27]: so what exactly is `ReactComponent` type

File: components/search_bar/search_bar.tsx [hunk 5]
```diff
@@ -7,8 +7,9 @@ import {FormattedMessage, useIntl} from 'react-intl';
 
 import Constants from 'utils/constants';
 import * as Utils from 'utils/utils';
+
+import SuggestionDate from 'components/suggestion/suggestion_date';
 import SearchSuggestionList from 'components/suggestion/search_suggestion_list.jsx';
-import SuggestionDate from 'components/suggestion/suggestion_date.jsx';
 import SuggestionBox from 'components/suggestion/suggestion_box';
```
Comment (M-ZubairAhmed) [inline components/search_bar/search_bar.tsx:13]: nit : move back in import is same position

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: components/search_bar/search_bar.tsx (hunk 5 above)
Comment (BediNimret) [reply components/search_bar/search_bar.tsx:13]: okay

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @BediNimret 
Tested, looks good to merge.
[E2E report](https://automation-dashboard.vercel.app/cycles/295e5121-58bd-49f0-9bbe-a788726c46d4) - no PR related failures.
